### ISSUE TITLE
errors with manual search

### ISSUE THREAD

**Issue by rmichelena** [morpheus65535/bazarr#1278]
A show for which there probably are no subtitles available.
(En Thérapie (2020), French) - manual search for English or Spanish subtitles

Screenshot of error attached

**Software (please complete the following information):**
 - Bazarr: 0.9.1
 - OS: Ubuntu
<img width="1267" alt="Screen Shot 2021-02-16 at 17 15 05" src="https://user-images.githubusercontent.com/77797875/108400364-1d6e0400-71e9-11eb-9b40-27c117a6ea69.png">

**Comment by morpheus65535** [comment]
Using browser dev console, please provide the response to this API endpoint call: `/api/episodes_subtitles_manual_search`

Also, please provide a debug log.

**Comment by rmichelena** [comment]
absolutely no idea how to do that from here (home) to my Bazarr which is on a docker container on a remote VPS.
is there a way to get what you need from the bash console of the container itself? curl or something?

**Comment by morpheus65535** [comment]
For browser dev console, here's a video: https://www.youtube.com/watch?v=Kkv30vZyQ14

Regarding debug log: https://github.com/morpheus65535/bazarr/wiki/FAQ#how-to-report-a-problem

Nothing here you can't do remotely.

**Comment by rmichelena** [comment]
<img width="1213" alt="Screen Shot 2021-02-26 at 17 28 32" src="https://user-images.githubusercontent.com/77797875/109361964-159d0800-7858-11eb-9315-f79cd0d9ecad.png">

**Comment by morpheus65535** [comment]
If you click on the first row link, can you see the raw response?

**Comment by rmichelena** [comment]
{"message": "The method is not allowed for the requested URL."}

**Comment by morpheus65535** [comment]
You are connecting trough a reverse proxy?

**Comment by rmichelena** [comment]
mmm... mine is a Cloudbox install, so I believe everything goes through nginx-proxy... but manual search was working fine a couple of versions ago...

**Comment by morpheus65535** [comment]
Manual search is working fine, maybe some issue with your DB. You didn't provide the requested debug log. I can't identify the issue without it.

**Comment by rmichelena** [comment]
[bazarr.log](https://github.com/morpheus65535/bazarr/files/6055316/bazarr.log)

emptied log, turned debug on, went to do a manual search, error appeared, turned debug off, downloaded log
as fast as I could, nevertheless it generated 15 pages of logs

**Comment by morpheus65535** [comment]
The request is taking too long to be rendered by Datatables that timeout (it takes almost 1 minute and timeout is 30 seconds). You can try to search for providers simultaneously, reduce the number of providers or deploy Bazarr on faster hardware.

**Comment by rmichelena** [comment]
mmm.. actually search simultaneously was enabled.
disabled it and the error went away.

**Comment by rmichelena** [comment]
also seems to be the search was much faster when I disabled search simultaneously - maybe because it didn't cause a timeout, or maybe the toggle is inverted?

**Comment by morpheus65535** [comment]
Toggle should be fine... Our log show clearly that they are called once after the other... really strange. It shouldn't be faster when calling them once after the other unless you're running on really slow hardware that can't deal with concurrent connections.

**Comment by rmichelena** [comment]
well that log I sent you was with "search simultaneously" enabled... and you say the logs show them being called sequentially?

**Comment by rmichelena** [comment]
while it's a VPS (Hetzner 2-processor), never had a problem that pointed to it being too slow... it even does sub sync without trouble

**Comment by rmichelena** [comment]
weird... not sure but seems to be related to having bsplayer and supersubtitles providers enabled
I'm managing to get it to work now with and without "simultaneously" enabled, as long as I have disabled bsplayer and supersubtitles, and have restarted the container
(bsplayer fills the logs with connection problems, and keeps filling even after disabling it, unless I restart the container)

**Comment by rmichelena** [comment]
maybe you noticed in my logs, bsplayer connection errors spam the logs endlessly. So I had disabled them for the last try.
I re-enabled bsplayer and supersubtitles, and the error came back; disabled them and the error was still there. restarted the container and it's fine again (with those providers disabled).
I've also previously noted that when I disable bsplayer, the logs keep filling with connection attempts to it, unless I restart the container.

**Comment by rmichelena** [comment]
re-enabled those providers and the error came back
the reason I chose to experiment disabling those two providers was that in the logs I sent you, besides the fact that bsplayer fills the logs with connection problems, those two also give an "unexpected error" with traceback and all, so it seemed to me they were crashing something
maybe the code that manages bsplayer is at fault, opening too many connections, and making everything time out? it's the only provider which fills the logs that way

**Comment by rmichelena** [comment]
this is the log part I was talking about:

27/02/2021 22:15:23|ERROR   |subliminal_patch.core           |Unexpected error in provider 'bsplayer': Traceback (most recent call last):  File "/app/bazarr/../libs/subliminal_patch/core.py", line 170, in list_subtitles_provider    results = self[provider].list_subtitles(video, provider_languages)  File "/app/bazarr/../libs/subliminal_patch/core.py", line 104, in __getitem__    provider = provider_registry[name](**self.provider_configs.get(name, {}))  File "/app/bazarr/../libs/subliminal_patch/providers/bsplayer.py", line 111, in __init__    self.initialize()  File "/app/bazarr/../libs/subliminal_patch/providers/bsplayer.py", line 115, in initialize    self.search_url = self.get_sub_domain()  File "/app/bazarr/../libs/subliminal_patch/providers/bsplayer.py", line 263, in get_sub_domain    raise ServiceUnavailable()subliminal.exceptions.ServiceUnavailable|Traceback (most recent call last):  File "/app/bazarr/../libs/subliminal_patch/core.py", line 170, in list_subtitles_provider    results = self[provider].list_subtitles(video, provider_languages)  File "/app/bazarr/../libs/subliminal_patch/core.py", line 104, in __getitem__    provider = provider_registry[name](**self.provider_configs.get(name, {}))  File "/app/bazarr/../libs/subliminal_patch/providers/bsplayer.py", line 111, in __init__    self.initialize()  File "/app/bazarr/../libs/subliminal_patch/providers/bsplayer.py", line 115, in initialize    self.search_url = self.get_sub_domain()  File "/app/bazarr/../libs/subliminal_patch/providers/bsplayer.py", line 263, in get_sub_domain    raise ServiceUnavailable()subliminal.exceptions.ServiceUnavailable|

**Comment by morpheus65535** [comment]
BSplayer is probably one of the worst providers we have developed in Bazarr. I've tried to improve it but they are really picky and for some it works perfectly fine while for others they always drop connection. In your case, having it running from a seedbox isn't gonna help you. They are heavily throttling connection and it looks like they aren't being really nice with shared IP (like seedbox). I suggest you disable this provider if you don't get good results out of it.

In my case, I've just run 20 search in line without a single timeout or connection refused. Works like a charm for me right now but experience may vary...
